bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusGroupHandler.java

```diff
@@ -0,0 +1,154 @@
+        /*
+         * Group Id has moved from config to property. Copy for backward compatibility
+         */
+        if (getThing().getProperties().get(CBusBindingConstants.PROPERTY_GROUP_ID) == null) {
+            if (getConfig().get(CBusBindingConstants.CONFIG_GROUP_ID) != null) {
+                updateProperty(CBusBindingConstants.PROPERTY_GROUP_ID,
+                        getConfig().get(CBusBindingConstants.CONFIG_GROUP_ID).toString());
+            }
+        }
+        if (getThing().getProperties().get(CBusBindingConstants.PROPERTY_GROUP_NAME) == null) {
+            if (getConfig().get(CBusBindingConstants.CONFIG_NAME) != null) {
+                updateProperty(CBusBindingConstants.PROPERTY_GROUP_NAME,
+                        getConfig().get(CBusBindingConstants.CONFIG_NAME).toString());
+            }
+        }
+        groupId = Integer.parseInt(getThing().getProperties().get(CBusBindingConstants.PROPERTY_GROUP_ID));
+        cBusNetworkHandler = getCBusNetworkHandler();
+        if (cBusNetworkHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            return;
+        }
+        updateStatus();
+    }
+
+    @SuppressWarnings({ "null" })
+    public void updateStatus() {
+        try {
+            logger.debug("updateStatus {}", getThing().getProperties().get(CBusBindingConstants.PROPERTY_GROUP_ID));
+            CBusNetworkHandler networkHandler = cBusNetworkHandler;
+            if (networkHandler == null || !networkHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            } else {
+                Group group = this.group;
+                if (group == null) {
+                    @Nullable
+                    Object groupId = getThing().getProperties().get(CBusBindingConstants.PROPERTY_GROUP_ID);
+                    if (groupId != null) {
+                        group = getGroup(Integer.parseInt(groupId.toString()));
+                        this.group = group;
+                    }
+                }
+                if (group == null) {
+                    logger.debug("Set state to configuration error -no group");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                } else if (group.getNetwork().isOnline()) {
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Network is not reporting online");
+                }
+            }
+        } catch (CGateException e) {
+            logger.warn("Problem checking network state for network {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    public void updateGroup(int application, int group, String value) {
+    }
```

Comment:
If this must be implemented by subclasses then it would be better to mark it as abstract.
```suggestion
    public abstract void updateGroup(int application, int group, String value);
```

Reply:
Done.